pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -461,25 +482,105 @@ pub mod pallet {
 
 						Ok::<(), DispatchError>(())
 					}
-				}
-			})?;
+				},
+			)?;
 
 			let expected_proof_count = Self::get_expected_proof_count(&router_ids)?;
 
 			Self::execute_if_requirements_are_met(
-				proof,
+				message_hash,
 				&router_ids,
 				session_id,
 				expected_proof_count,
 				domain_address,
 			)?;
 
-			Self::deposit_event(Event::<T>::MessageRecoveryExecuted { proof, router_id });
+			Self::deposit_event(Event::<T>::MessageRecoveryExecuted {
+				message_hash,
+				router_id,
+			});
+
+			Ok(())
+		}
+
+		/// Sends a message that initiates a message recovery using the
+		/// messaging router.
+		///
+		/// Can only be called by `AdminOrigin`.
+		#[pallet::weight(T::WeightInfo::initiate_message_recovery())]
+		#[pallet::call_index(12)]
+		pub fn initiate_message_recovery(
```

Comment:
cc @hieronx

Reply:
This looks right to me!

Only note: `Adminorigin` is our LP admin right? This should be `ensure_root`, like `Schedule/CancelUpgrade`.

Reply:
It's:

https://github.com/centrifuge/centrifuge-chain/blob/d1d823acbb9dbc0a1d7ac0aa7679ee30b34320c7/runtime/centrifuge/src/lib.rs#L1865

Reply:
That should work too!

Reply:
Just to be 100% sure since you originally did not want to give that power to the LpAdmin: This includes three parties:
1. `Root`
2. 2/3 of council
3. Our LPAdminAcc

@hieronx

Reply:
I think that is fine for now. We should probably remove the LPAdmin in the next 6 months or so and just use `Root` for all of this. But until then it works as is.

Reply:
In 6 months we should be using OpenGov and then have the `Pool` track for those type of things.